src/effects/RealtimeEffectList.cpp

```diff
@@ -104,11 +104,37 @@ void RealtimeEffectList::RemoveState(RealtimeEffectState &state)
       mStates.erase(found);
 }
 
+size_t RealtimeEffectList::GetStatesCount() const noexcept
+{
+   return mStates.size();
+}
+
+RealtimeEffectState* RealtimeEffectList::GetStateAt(size_t index) noexcept
+{
+   return mStates[index].get();
+}
+
 void RealtimeEffectList::Swap(size_t index1, size_t index2)
 {
    std::swap(mStates[index1], mStates[index2]);
 }
 
+void RealtimeEffectList::Reorder(size_t fromIndex, size_t toIndex)
+{
+   assert(fromIndex < mStates.size());
+   assert(toIndex <= mStates.size());
+   if(fromIndex != toIndex)
+   {
+      auto effectState = std::move(mStates[fromIndex]);
+      mStates.erase(mStates.begin() + fromIndex);
+      
+      mStates.emplace(
+         mStates.begin() + (toIndex < fromIndex ? toIndex : toIndex - 1),
+         std::move(effectState)
+      );
+   }
```

Comment:
Use `std::rotate` to simplify this

Reply:
And becuase this does all that `Swap` can do, and more -- then remove `Swap`

Reply:
I'll skip this, as you've already done that